src/manager.js

```diff
@@ -0,0 +1,166 @@
+import picoSignals from 'pico-signals';
+import createPGroup from './util/p-group';
+
+const findLocaleById = (locales, localeId) => {
+    const locale = locales.find(({ id }) => id === localeId);
+
+    if (!locale) {
+        throw new Error(`Unknown locale id: ${localeId}`);
+    }
+
+    return locale;
+};
+
+const match = (locales, policies, ctx = {}) => {
+    const localeId = policies.reduce((localeId, policy) => localeId ?? policy.match(locales, ctx), null);
+
+    if (!localeId) {
+        throw new Error('None of the policies matched a locale.. did you forgot to include the default policy?');
+    }
+
+    return findLocaleById(locales, localeId);
+};
+
+const save = async (policies, locale) => {
+    const policy = policies.find((policy) => policy.save);
+
+    if (policy) {
+        await policy.save(locale);
+    }
+};
+
+const act = (policies, locale) => {
+    let acted = false;
+
+    return policies.reduce((suspendAct, policy) => {
+        if (!acted && policy.act) {
+            suspendAct = policy.act(locale);
+            acted = true;
+        }
+
+        return suspendAct;
+    }, null);
+};
+
+const watch = (policies, callback) => {
+    const suspendWatchFns = policies.reduce((suspendWatchFns, policy) => {
+        if (policy.watch) {
+            suspendWatchFns.push(policy.watch(callback));
+        }
+
+        return suspendWatchFns;
+    }, []);
+
+    return () => {
+        suspendWatchFns.forEach((suspend) => suspend());
+    };
+};
+
+const getInitialData = async (locales, policies, ctx) => {
+    if (typeof window !== 'undefined') {
+        return;
+    }
+
+    const locale = match(locales, policies, ctx);
+    const messages = await locale.loadMessages();
+
+    return {
+        localeId: locale.id,
+        messages,
+    };
+};
+
+const createManager = (locales, policies, initialData) => {
+    let locale = findLocaleById(locales, initialData.localeId);
+    let messages = initialData.messages;
+
+    const changeSignal = picoSignals();
+    const pGroup = createPGroup();
+    let suspendWatch;
+    let suspendAct;
+
+    /* istanbul ignore else */
```

Comment:
🔔 shame

Why don't you just test that the API is still exported when it's running in SS?

Reply:
Lazyness

Reply:
Added a test that covers this.